distributed/actor.py

```diff
@@ -59,12 +59,14 @@ def __init__(self, cls, address, key, worker=None):
             self._client = None
         else:
             try:
+                # TODO: `get_worker` may return the wrong worker instance for async local clusters (most tests)
```

Comment:
If this is running in a task (normal or actor), then the thread-local execution state must contain the right worker, no await would have happened, no? You're thinking of cases where the task itself is async?

See [this line](https://github.com/dask/distributed/pull/4287/files#diff-095a1af29784e71235a0a6eaf77169c7b1c9eff5cb5cf84d69323dedb880805eR64) in my related unmerged PR.